### ISSUE TITLE
"Cannot set property 'noEmit' of undefined"

### ISSUE THREAD

**Issue by dwickern** [emberwatch/ember-cli-typescript#43]
I get this error when trying to use e-c-ts in a fresh addon. Maybe I'm doing something wrong. Here's what I did:
1. Created my addon: `ember addon my-addon --yarn`
1. Installed e-c-ts: `ember install ember-cli-typescript`
1. Moved the e-c-ts dependency from `devDependencies` into `dependencies` as documented in the README
1. `yarn link`ed the dependency into my app and tried to run it

It seems that it's trying to load the tsconfig from the root of my app instead of from the addon
```
Cannot set property 'noEmit' of undefined
TypeError: Cannot set property 'noEmit' of undefined
    at TypeScriptPreprocessor.toTree (/Users/dwickern/code/my-addon/node_modules/ember-cli-typescript/lib/typescript-preprocessor.js:68:37)
    at /Users/dwickern/code/my-app/node_modules/ember-cli-preprocess-registry/preprocessors.js:180:26
    at Array.forEach (native)
    at processPlugins (/Users/dwickern/code/my-app/node_modules/ember-cli-preprocess-registry/preprocessors.js:178:11)
    at Function.module.exports.preprocessJs (/Users/dwickern/code/my-app/node_modules/ember-cli-preprocess-registry/preprocessors.js:171:10)
    at Class.preprocessJs (/Users/dwickern/code/my-app/node_modules/ember-cli/lib/models/addon.js:1085:25)
    at Class.processedAddonJsFiles (/Users/dwickern/code/my-app/node_modules/ember-cli/lib/models/addon.js:1099:33)
    at Class.compileAddon (/Users/dwickern/code/my-app/node_modules/ember-cli/lib/models/addon.js:1007:24)
    at Class.treeForAddon (/Users/dwickern/code/my-app/node_modules/ember-cli/lib/models/addon.js:661:26)
    at Class._treeFor (/Users/dwickern/code/my-app/node_modules/ember-cli/lib/models/addon.js:517:33)
```

**Comment by mfeckie** [comment]
I have had the same issue and 'fixed' it like this https://github.com/mfeckie/ember-cli-typescript/commit/f1ce57a5e0a47ef43ecc49941f4ea46ce3faa733

**Comment by chriskrycho** [comment]
This should be resolved entirely by #45. 🎉

**Comment by chriskrycho** [comment]
This should be resolved in the 1.0 beta. Please test with `"ember-cli-typescript": "next"` and let me know if it recurs.
